snap/snapcraft.yaml

```diff
@@ -133,6 +153,38 @@ parts:
     stage-packages:
       - inotify-tools
 
+  cuda-samples:
+    source: https://github.com/NVIDIA/cuda-samples
+    source-tag: v11.5
+    source-type: git
+    source-depth: 1
+    plugin: nil
+    build-packages:
+      - build-essential
+      - on amd64:
+        - libnvidia-compute-515-server
+        - nvidia-cuda-toolkit
+      - on arm64:
+        - libnvidia-compute-515-server
+        - nvidia-cuda-toolkit
+    override-build: |
+      case $CRAFT_TARGET_ARCH in
+      amd64|arm64)
+      make -C Samples/bandwidthTest CUDA_PATH=/usr/lib/nvidia-cuda-toolkit
+      make -C Samples/deviceQuery CUDA_PATH=/usr/lib/nvidia-cuda-toolkit
+      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/bin/
+      cp bin/*/linux/release/* $SNAPCRAFT_PART_INSTALL/usr/bin/
+      ;;
+      *)
+      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/bin/
+      cp /bin/false $SNAPCRAFT_PART_INSTALL/usr/bin/bandwidthTest
+      cp /bin/false $SNAPCRAFT_PART_INSTALL/usr/bin/deviceQuery
```

Comment:
Could these be symlinks instead?

Reply:
review tools usually rejects broken symlinks or symlinks pointing outside of the snap. Unless this has changed.....

Reply:
Even better, make these `echo 'This only works on amd64 and arm64'; false`?